### PR TITLE
Bump aks-engine to v0.31.1 to support private ACR

### PR DESCRIPTION
**Pull request by ritazh**
Incorporate aks-engine update: https://github.com/Azure/aks-engine/pull/523

**Notes**:

```release-note
Update tests and docs to use aks-engine v0.31.1 to support private ACR to fix e2e tests
```

### CODE REVIEW

File: tests/k8s-azure/manifest/linux-kcm.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
             "orchestratorRelease": "1.14",
             "kubernetesConfig": {
                 "customHyperkubeImage": "gcrio.azureedge.net/google_containers/hyperkube-amd64:v1.13.0-alpha.3",
+                "privateAzureRegistryServer": "gcrio.azureedge.net",
```
Comment (andyzhangx) [inline tests/k8s-azure/manifest/linux-kcm.json]: actually it's gcr.io, not azure registry...

Comment (ritazh) [reply tests/k8s-azure/manifest/linux-kcm.json]: @andyzhangx This value will be replaced by the generated image that is being pushed to ACR.

**Pull request merged**